.evergreen/scripts/run-auth-tests.sh

```diff
@@ -100,12 +100,15 @@ elif command -v otool >/dev/null; then
 fi
 
 if [[ "${ssl}" != "OFF" ]]; then
+  if true; then
```

Comment:
Recommend using `{} || {}` instead of `if/else` to better distinguish the conditional blocks from existing code:

```bash
{
  # Skip 3.6. Remove this block when resolving CDRIVER-5645.
  echo "Skipping tests on server 3.6 (using 'auth_host' or 'auth_gssapi') until DEVPROD-9029 is addressed."
} || {
# FIXME: CDRIVER-2008 for the cygwin check
if [[ "${OSTYPE}" != "cygwin" ]]; then
  echo "Authenticating using X.509"
  ...
fi
}
```